src/gui/dialog/SettingsDialog.cpp

```diff
@@ -269,6 +270,47 @@ void SettingsDialog::load()
 	int timeoutMs = 1000;
 	touch.getInt("timeout", timeoutMs);
 	gtk_spin_button_set_value(GTK_SPIN_BUTTON(get("spTouchDisableTimeout")), timeoutMs / 1000.0);
+
+    this->audioInputDevices = this->control->getAudioController()->getAudioRecorder()->getInputDevices();
+    for (auto &audioInputDevice : this->audioInputDevices)
+	{
+    	gtk_combo_box_text_append(GTK_COMBO_BOX_TEXT(get("cbAudioInputDevice")), "", audioInputDevice.getDeviceName().c_str());
+	}
+    for (int i = 0; i < this->audioInputDevices.size(); i++)
+    {
+    	if (this->audioInputDevices[i].getSelected())
+		{
+			gtk_combo_box_set_active(GTK_COMBO_BOX(get("cbAudioInputDevice")), i);
+		}
+    }
+
+    this->audioOutputDevices = this->control->getAudioController()->getAudioPlayer()->getOutputDevices();
+	for (auto &audioOutputDevice : this->audioOutputDevices)
+	{
+		gtk_combo_box_text_append(GTK_COMBO_BOX_TEXT(get("cbAudioOutputDevice")), "", audioOutputDevice.getDeviceName().c_str());
+	}
+	for (int i = 0; i < this->audioOutputDevices.size(); i++)
+	{
+		if (this->audioOutputDevices[i].getSelected())
+		{
+			gtk_combo_box_set_active(GTK_COMBO_BOX(get("cbAudioOutputDevice")), i);
+		}
+	}
+
+	switch((int)settings->getAudioSampleRate())
+	{
+		default:
```

Comment:
Usually default should be at the end, even modern compiler support it may this way (this is no blocker!)